### ISSUE TITLE
jdmpview info mod shows <no section information> for Java modules

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#9549]
Running the jdmpview `info mod` command on [jdk8u252 for plinux](https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u252-b09_openj9-0.20.0/OpenJDK8U-jdk_ppc64le_linux_openj9_8u252b09_openj9-0.20.0.tar.gz) results in `<no section information>` output for Java modules, as shown below (not a full list). This affects using DTJF to read symbols from these modules.

jdk11 has the same problem, likely other versions as well.

Possibly this is related to the following changes:
https://github.com/ibmruntimes/openj9-openjdk-jdk8/commit/58e0fd0
https://github.com/ibmruntimes/openj9-openjdk-jdk8/commit/3902bbd
https://github.com/ibmruntimes/openj9-openjdk-jdk8/commit/1a074c8

```
	  libnet.so @ 0x3fff8cfb0000, <no section information>

	  libnio.so @ 0x3fff8cff0000, <no section information>

	  libzip.so @ 0x3fff8d630000, <no section information>

	  libverify.so @ 0x3fff8d670000, <no section information>

	  libjava.so @ 0x3fff8d6a0000, <no section information>

	  libjclse29.so @ 0x3fffa4610000, <no section information>

	  libj9gc29.so @ 0x3fffa46c0000, <no section information>

	  libj9jvmti29.so @ 0x3fffb0080000, <no section information>

	  libj9shr29.so @ 0x3fffb0110000, <no section information>

	  libj9vrb29.so @ 0x3fffb01e0000, <no section information>

	  libj9trc29.so @ 0x3fffb0300000, <no section information>

	  libj9zlib29.so @ 0x3fffb0360000, <no section information>

	  libj9dmp29.so @ 0x3fffb03a0000, <no section information>
```

**Comment by pshipton** [comment]
I was able to go back to being able to read/display shared library info by removing the `-pie` from https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/openj9/common/autoconf/flags.m4#L819

I didn't have any luck making this change and doing a incremental build, but the behavior changed after a full build.

@keithc-ca I captured the JVMs and the cores, they are in team/keithc/openj9-9549. The files labelled `before` have the `<no section information>` behavior. These jvms are cores are from plinux (64-bit).

You could duplicate this problem on x86 for 32-bit only, by removing the x32 check https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/openj9/common/autoconf/flags.m4#L818

**Comment by keithc-ca** [comment]
It seems a difference is that the 'before' core file has no executable (type 2) sections (just the core (type 4) section followed by the shared object (type 3) sections):
```
$ hexdump -C core.before.dmp | grep -A1 '7f 45 4c 46'
00000000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
00000010  04 00 15 00 01 00 00 00  00 00 00 00 00 00 00 00  |................|
--
03a10000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
03a10010  03 00 15 00 01 00 00 00  f0 07 00 00 00 00 00 00  |................|
--
03a20000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
03a20010  03 00 15 00 01 00 00 00  f0 07 00 00 00 00 00 00  |................|
--
[...]
```
while the 'after' core file has the expected executable sections:
```
$ hexdump -C core.after.dmp | grep -A1 '7f 45 4c 46' | head -20
00000000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
00000010  04 00 15 00 01 00 00 00  00 00 00 00 00 00 00 00  |................|
--
03210000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
03210010  02 00 15 00 01 00 00 00  c0 05 00 10 00 00 00 00  |................|
--
03220000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
03220010  02 00 15 00 01 00 00 00  c0 05 00 10 00 00 00 00  |................|
--
07e30000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
07e30010  03 00 15 00 01 00 00 00  20 2c 00 00 00 00 00 00  |........ ,......|
--
08010000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
08010010  03 00 15 00 01 00 00 00  80 49 00 00 00 00 00 00  |.........I......|
--
[...]
```
Note the type is the 16-bit value at offset 0x10 of each section.

**Comment by keithc-ca** [comment]
With the help of `gdb`, I learned that in the 'before' core, the executable corresponds to the section at file offset 0x03a10000. As noted above, that is shared object (type 3) section. Hopefully, `jdmpview` can be fixed to recognize that section as the appropriate shared object.

**Comment by keithc-ca** [comment]
In the troublesome core files, it seems the section table of the executable has not been loaded or has been over-written: We need to find another source for that information.

**Comment by keithc-ca** [comment]
I made improvements to `jextract` so it can find more of the necessary files; running `jdmpview` in the context of those extra files seems to enable most section information. I'll be opening a pull request shortly.

**Comment by pshipton** [comment]
Now that jextract has been updated to find the java and system libraries, can jdmpview be updated with similar changes?

**Comment by keithc-ca** [comment]
Yes, it could. However, it will need more care because `jdmpview` shouldn't be making the assumption (as `jextract` does) that the core file was produced by the same JDK on the same system.
